pkg/generators/extension.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generators
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/golang/glog"
+	"k8s.io/gengo/types"
+)
+
+// Extension tag to openapi extension
+var tagToExtension = map[string]string{
+	"patchMergeKey": "x-kubernetes-patch-merge-key",
+	"patchStrategy": "x-kubernetes-patch-strategy",
+	"listType":      "x-kubernetes-list-type",
+	"listTypeKey":   "x-kubernetes-list-type-keys",
+}
+
+// Enum values per extension
+var allowedExtensionValues = map[string][]string{
+	"x-kubernetes-patch-strategy": []string{"merge", "retainKeys"},
```

Comment:
map[string]bool{} makes a loop below a 1-liner

Reply:
Seems like a premature optimization. The size of the lists are small. And it will be easier to print out the allowed values. We can revisit this in another PR.

Reply:
Not sure if I understand your comment here Daniel. Is this list a map from an extension to its allowed values? how map[string]bool can serve that purpose?